### PR TITLE
Allow @usableFromInline and @inlinable to package decls

### PR DESCRIPTION
**Pull request by elsh**
Allow @usableFromInline and @inlinable to package decls
Add tests 
Resolves rdar://104617133

This PR allows the following (similar to how the attributes works with internal/public decls): 

```
@usableFromInline package func foo() {}
@inlinable package func bar() { 
    foo() 
}
@inlinable public func baz() { 
    foo() 
}
```

### CODE REVIEW

File: test/attr/accessibility_proto.swift [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// RUN: %target-typecheck-verify-swift -package-name myPkg
```
Comment (elsh) [inline test/attr/accessibility_proto.swift:1]: Adding this back (removed in the previous PR); changes in this file are needed to match the original (before the PR)

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

**Review by xymus — APPROVED** [review]

File: test/Sema/accessibility_package_inline.swift [hunk 2]
```diff
@@ -0,0 +1,180 @@
+// RUN: %empty-directory(%t)
+// RUN: %{python} %utils/split_file.py -o %t %s
+
+// RUN: %target-swift-frontend -module-name UtilsGood %t/UtilsGood.swift -emit-module -emit-module-path %t/UtilsGood.swiftmodule -package-name myLib
+// RUN: test -f %t/UtilsGood.swiftmodule
+
+// RUN: not %target-swift-frontend -typecheck %t/Utils.swift -package-name myLib -I %t 2> %t/resultUtils.output
+// RUN: %FileCheck %s -input-file %t/resultUtils.output -check-prefix CHECK-UTILS
+// CHECK-UTILS: error: class 'PackageKlass' is package and cannot be referenced from an '@inlinable' function
+// CHECK-UTILS:   let a = PackageKlass()
```
[the remark this answers is not in the document]
Comment (elsh) [reply test/Sema/accessibility_package_inline.swift:10]: Makes sense; will update the test in the next PR

**Pull request merged**